### PR TITLE
Address preview state in ConnectEhrStep and ConnectDeviceAccountStep

### PR DESCRIPTION
**Pull request by bxh**
## Overview

- Display preview state in ConnectEhrStep when in design mode.
- Require `deviceType` in ConnectDeviceAccountStep.

Address https://github.com/CareEvolution/MyDataHelpsDesignerUI/issues/350

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [x] [MyDataHelps website](mydatahelps.org)

### Documentation
- [ ] I have added relevant Storybook updates to this PR as well.
- [ ] If this feature requires a developer doc update, tag @CareEvolution/api-docs.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

### CODE REVIEW

**Review by ashleyrtate — APPROVED** [review]
Just one minor suggestion, otherwise 🚢

File: src/components/step/ConnectEhrStepContainer/ConnectEhrStepContainer.tsx [hunk 1]
```diff
@@ -40,6 +49,8 @@ export default function () {
             nextButtonDisabled={nextButtonDisabled}
             styles={styles}
             onProviderConnected={onProviderConnected}
+            // Display preview data under 'Design' mode.
+            previewState={surveyContext?.surveyMode === 'Design'?'Default' : undefined}
```
[the remark this answers is not in the document]
Comment (bxh) [reply src/components/step/ConnectEhrStepContainer/ConnectEhrStepContainer.tsx:53]: `ProviderSearchPreviewState` is a type not a constant.

P.S: sorry for ignoring this thread for this long. @ashleyrtate

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 2]
```diff
@@ -2,17 +2,18 @@ import React, { useState, useEffect } from "react";
 import MyDataHelps, { StepConfiguration } from "@careevolution/mydatahelps-js";
 import ConnectDeviceAccountStep from "../ConnectDeviceAccountStep";
 import { getFitbitProviderID, getGarminProviderID, getOmronProviderID } from "../../../helpers/providerIDs";
+import { deviceType } from "../../../helpers/deviceType";
 
 export interface ConnectDeviceAccountStepContainerProps {
-    deviceType?: string;
+    deviceType: deviceType
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:8]: @bxh why is this required now?  Isn't device type set by step configuration?  So what should this be set to?

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:8]: To be clear, view library isn't building because we put this step container in view library.  It should be inferring it's configuration from the SDK call to get configuration, we don't set device type in the view library

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:8]: `deviceType` is `Fitbit`, `Garmin`, or `Omron`. Do you mean the value can be read from the step configuration?

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:8]: The value is always supposed to be read from step configuration...so it doesn't make sense to make this prop required

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:8]: OK, can I fix it next week?

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:8]: I already fixed it: https://github.com/CareEvolution/MyDataHelpsUI/pull/270
